pkg/util/acrtoken/acrtoken.go

```diff
@@ -98,14 +99,69 @@ func (m *manager) EnsureTokenAndPassword(ctx context.Context, rp *api.RegistryPr
 		return "", err
 	}
 
+	return m.generateTokenPassword(ctx, mgmtcontainerregistry.TokenPasswordNamePassword1, rp)
+}
+
+// RotateTokenPassword chooses either the unused token password or the token
+// password with the oldest creation date, generates a new password, and
+// then updates the registry profile with the newly generated password.
+func (m *manager) RotateTokenPassword(ctx context.Context, rp *api.RegistryProfile) error {
+	tokenProperties, err := m.tokens.GetTokenProperties(ctx, m.r.ResourceGroup, m.r.ResourceName, rp.Username)
+	if err != nil {
+		return err
+	}
+	tokenPasswords := *tokenProperties.Credentials.Passwords
+
+	var passwordToRenew mgmtcontainerregistry.TokenPasswordName
+	switch {
+	// Passwords only has one entry: renew password that isn't present
+	case len(tokenPasswords) == 1:
+		if tokenPasswords[0].Name == mgmtcontainerregistry.TokenPasswordNamePassword1 {
+			passwordToRenew = mgmtcontainerregistry.TokenPasswordNamePassword2
+		} else {
+			passwordToRenew = mgmtcontainerregistry.TokenPasswordNamePassword1
+		}
+	// Passwords has two entries: compare creation dates, renew oldest
+	case len(tokenPasswords) == 2:
+		if tokenPasswords[0].CreationTime.Before(tokenPasswords[1].CreationTime.Time) {
+			passwordToRenew = mgmtcontainerregistry.TokenPasswordNamePassword1
+		} else {
+			passwordToRenew = mgmtcontainerregistry.TokenPasswordNamePassword2
+		}
+	// default case, including passwords having zero entries: generate password 1
+	// this shouldn't ever happen, which guarantees it will happen
+	default:
+		passwordToRenew = mgmtcontainerregistry.TokenPasswordNamePassword1
+	}
```

Comment:
As it stands the shared functionality of both workflows is already in `generateTokenPassword`, which takes the password name as an argument of which password to generate. The logic here is just figuring out which password on the token should be used, while `EnsureTokenAndPassword` has the advantage of just being able to have a static password name to generate since the token is newly created. Unifying the functionality any more than it already is would probably mean using the registry token itself as a function argument and having the logic to choose the password inside that function. The problem there is that the `CreateAndWait` function used on line 88 only returns an error, not the entire token. That function would either have to be changed to return the token on success (which may not be a bad idea), or the token would need to be retrieved separately.